code/renderers/react/src/renderToCanvas.tsx

```diff
@@ -81,12 +98,22 @@ export async function renderToCanvas(
     unmountElement(canvasElement);
   }
 
-  await act(async () => {
-    await renderElement(element, canvasElement, storyContext?.parameters?.react?.rootOptions);
+  await new Promise<void>((resolve, reject) => {
+    try {
+      actQueue.push(async () => {
+        await act(async () => {
+          await renderElement(element, canvasElement, storyContext?.parameters?.react?.rootOptions);
+          resolve();
+        });
+      });
+      processActQueue();
+    } catch (e) {
+      reject(e);
+    }
   });
 
-  return async () => {
-    await act(() => {
+  return () => {
+    act(() => {
```

Comment:
this change now means that whatever `act()` returns or throws is ignored. is that on purpose?

Reply:
Indeed, I will revert this change because React's recommendation is to await every `act` call since React 19, and awaiting act doesn't harm in previous versions of React.